crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -263,22 +264,38 @@ def to_databag_hash
     private
 
     def repos_path
-      Pathname.new(File.join("/srv/tftpboot", @platform, @arch, "repos"))
+      Pathname.new("/srv/tftpboot").join(@platform, @arch, "repos")
+    end
+
+    def repo_path
+      repos_path.join(name)
+    end
+
+    def smt_path
+      unless @config["smt_path"].blank?
+        Pathname.new("/srv/www/htdocs/repo").join(@config["smt_path"])
+      end
     end
 
     def repodata_path
-      repos_path.join(name, "repodata")
+      repo_path.join("repodata")
     end
 
     def repodata_media_path
-      repos_path.join(name, "suse", "repodata")
+      repo_path.join("suse", "repodata")
+    end
+
+    def ensure_link_smt_path
+      unless remote? || smt_path.nil? || repo_path.directory? || !smt_path.directory?
```

Comment:
@toabctl yes, that's the `smt_path.directory?` bit. Although, this might not be enough if the symlink is dangling. Not sure if we should care about that, though?